op-e2e/config/init.go

```diff
@@ -144,69 +171,370 @@ func init() {
+
+	for _, allocType := range allocTypes {
+		if allocType == AllocTypeL2OO {
+			continue
 		}
-		if err != nil {
-			panic(err)
+
+		initAllocType(root, allocType)
+	}
+
+	configPath := path.Join(root, "op-e2e", "config")
+	forks := []genesis.L2AllocsMode{
+		genesis.L2AllocsHolocene,
+		genesis.L2AllocsGranite,
+		genesis.L2AllocsFjord,
+		genesis.L2AllocsEcotone,
+		genesis.L2AllocsDelta,
+	}
+
+	l2OOAllocsL1B := decompressGzipFile(path.Join(configPath, "allocs-l1.json.gz"))
+	var l2OOAllocsL1 foundry.ForgeAllocs
+	if err := json.Unmarshal(l2OOAllocsL1B, &l2OOAllocsL1); err != nil {
+		panic(fmt.Errorf("failed to unmarshal L2OO L1 allocs: %w", err))
+	}
+	l1AllocsByType[AllocTypeL2OO] = &l2OOAllocsL1
+
+	l2OOAddressesB := decompressGzipFile(path.Join(configPath, "addresses.json.gz"))
+	var l2OOAddresses genesis.L1Deployments
+	if err := json.Unmarshal(l2OOAddressesB, &l2OOAddresses); err != nil {
+		panic(fmt.Errorf("failed to unmarshal L2OO addresses: %w", err))
+	}
+	l1DeploymentsByType[AllocTypeL2OO] = &l2OOAddresses
+
+	l2OODC := DeployConfig(AllocTypeStandard)
+	l2OODC.SetDeployments(&l2OOAddresses)
+	deployConfigsByType[AllocTypeL2OO] = l2OODC
+
+	l2AllocsByType[AllocTypeL2OO] = genesis.L2AllocsModeMap{}
+	for _, fork := range forks {
+		l2OOAllocsL2B := decompressGzipFile(path.Join(configPath, fmt.Sprintf("allocs-l2-%s.json.gz", fork)))
+		var l2OOAllocsL2 foundry.ForgeAllocs
+		if err := json.Unmarshal(l2OOAllocsL2B, &l2OOAllocsL2); err != nil {
+			panic(fmt.Errorf("failed to unmarshal L2OO L2 allocs: %w", err))
 		}
+		l2AllocsByType[AllocTypeL2OO][fork] = &l2OOAllocsL2
 	}
-	if missing {
-		log.Warn("allocs file not found, skipping", "allocType", allocType)
-		return
+
+	// Use regular level going forward.
+	oplog.SetGlobalLogHandler(handler)
+}
+
+func initAllocType(root string, allocType AllocType) {
+	artifactsPath := path.Join(root, "packages", "contracts-bedrock", "forge-artifacts")
+	if err := ensureDir(artifactsPath); err != nil {
+		panic(fmt.Errorf("invalid artifacts path: %w", err))
 	}
 
-	l1Allocs, err := foundry.LoadForgeAllocs(l1AllocsPath)
+	loc, err := artifacts.NewFileLocator(artifactsPath)
 	if err != nil {
-		panic(err)
+		panic(fmt.Errorf("failed to create artifacts locator: %w", err))
 	}
-	l1AllocsByType[allocType] = l1Allocs
+
+	lgr := log.New()
+
+	allocModes := []genesis.L2AllocsMode{
+		genesis.L2AllocsHolocene,
+		genesis.L2AllocsGranite,
+		genesis.L2AllocsFjord,
+		genesis.L2AllocsEcotone,
+		genesis.L2AllocsDelta,
+	}
+
 	l2Alloc := make(map[genesis.L2AllocsMode]*foundry.ForgeAllocs)
-	mustL2Allocs := func(mode genesis.L2AllocsMode) {
-		name := "allocs-l2-" + string(mode)
-		allocs, err := foundry.LoadForgeAllocs(filepath.Join(l2AllocsDir, name+".json"))
-		if err != nil {
-			panic(err)
-		}
-		l2Alloc[mode] = allocs
+	var wg sync.WaitGroup
+
+	// Corresponds with the Deployer address in cfg.secrets
+	pk, err := crypto.HexToECDSA("7c852118294e51e653712a81e05800f419141751be58f605c371e15141b007a6")
```

Comment:
This could go stale easily, any way to have single source of truth?

Reply:
I'd like to refactor the way we do secrets more broadly.